torch/csrc/autograd/saved_variable.cpp

```diff
@@ -35,9 +35,7 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output) {
 
 Variable SavedVariable::unpack(std::shared_ptr<Function> saved_for) const {
   if (!data_.defined()) {
-    if (!was_default_constructed_) {
-      throw std::runtime_error(ERR_BACKWARD_TWICE);
-    }
+    AT_ASSERT(was_default_constructed_);
```

Comment:
Should probably be TORCH_INTERNAL_ASSERT. Also please add a comment or error message about why we can't reach this (we should have already checked if the buffer was released. If it was released, we should not be unpacking it.).